### PR TITLE
Scientists now randomly burst into flames

### PR DESCRIPTION
**Pull request by ghost**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl:
add: Lathes and Mechfabs now have a random chance to set things on fire around them, scaling quadratically with the part tier.
add: Engineers can now percussively maintain the machines to temporarily stop these malfunctions entirely
code: Added a function similar to `isInSight` called `isNotBlocked`, which just replaces `density` with `opacity`
/:cl:

## About The Pull Request
john cena nae nae

Changelog explains most of what this will bring. Science machinery will create random hotspots in a 7x7 radius (not through walls or solid object, I took care of that) when used. There is a risk of this every time a part is printed and is quite low, but it increases as you upgrade the machinery.

## Why It's Good For The Game
Scientists are the big homosex. This should add a good bit of **!!!FUN!!!** to their day-to-day games.

Note that science has more than enough tools to simply shrug off the flames, firesuits and extinguishers abound.

### CODE REVIEW

**Comment by Toxici11i** [comment]
unga dunga me want ruin science day because miners do job and everyone op...

**Comment by Toxici11i** [comment]
As a non shitpost comment, this is a bad change for the sake of a bad change. Science is already over-reliant on other departments to get resources etc, and this just increases it.

Also wont this effect every protolathe, not just sci?

**Comment by ghost** [comment]
@Toxici11i silence, science main.

This isn't a bad change for the sake of a bad change, but simply a first step in the long line of steps to make dynamic fires a feature.

Right now, atmos never does anything resembling their role of 'firefighter' quite simply because there aren't any fires to fight. I plan to revamp this from the ground-up.

Either way, this is nothing more than an inconvenience for science - it is very easy to avoid the flames and there aren't going to be too many instances where they actually appear.

In terms of other lathes, this should not be a problem. Science would have to actually upgrade them first, you see.

**Comment by ghost** [comment]
let engineers kick the machines on harm intent to make them work better and not fail for a short while. epic PR +1

**Comment by ghost** [comment]
@zamolxius that sounds like a plan

**Comment by yoyobatty** [comment]
Put that line of sight thing in another pr lol

**Comment by ghost** [comment]
@yoyobatty it's required for this to work properly.

**Comment by Neotwr** [comment]
+1, this can be hilarious

**Comment by shedowtnt** [comment]
i dont know if i like this or not
depends on how big is the fire
since fires are fucking annoying

**Comment by ghost** [comment]
@shedowtnt fires are random tiles in a 9x9 grid, but are blocked by solid objects like tables, windows, walls and doors. I tried it myself and it was very simple to find places in which you can completely avoid any and all fires.

Additionally, there are two fire lockers (three if you count the one in escape) in science and a chem dispenser down south. This is no more than a minor setback - unless you're near-crit, this shouldn't be a problem.

**Comment by mkikt4743** [comment]
+1, more problems to fight on slow rounds is good, also being a fire man is fun

**Comment by Manly-Dwarf** [comment]
I love workplace accidents.

**Comment by ghost** [comment]
@zamolxius boot of life is complete and tested

**Comment by ghost** [comment]
percussive maintenance. fucking epic

**Comment by GuyonBroadway** [comment]
Why limit this to R&D? make EVERY machine have a chance to cause an electrical fire.

Heck make it so all machines have a higher chance to do that the more charge is in the powernet, make engine funny!

**Comment by ghost** [comment]
@GuyonBroadway Now is not the time for more features.

That comes later.

**Comment by cris5311** [comment]
+1
finally the science nerf we need!

**Comment by ghost** [comment]
seems like this has been designer and HA approved, just waiting for review now

**Comment by Maroonenahmoron** [comment]
Not only it should make fires, but the machine should make odd noises and give some warning/indication that something is about to blow.

**Comment by ghost** [comment]
@Maroonenahmoron it sends a message into the chat and makes a distinctive sparking sound. There is then a 1.5 second delay, which is enough for a scientist to run out of the way.

**Comment by xTrainx** [comment]
at least the monthly pr shitting on science are getting honest now

**Comment by FantasticFwoosh** [comment]
``Have you tried turning the APC on and off again?``

**Comment by ExpAntique** [comment]
+1 on this and the idea of implementing more ways for science to go completely haywire. Maybe the ability for toxins to create gas leaks and R&D to create affronts to god.

**Comment by DON-KARMA** [comment]
Sims 4: Sci-Fi Expansion

**Comment by Neotwr** [comment]
merge when

**Review by ghost — APPROVED** [review]

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -315,7 +323,7 @@ Class Procs:
 		user.changeNext_move(CLICK_CD_MELEE)
 		user.do_attack_animation(src, ATTACK_EFFECT_PUNCH)
 		user.visible_message("<span class='danger'>[user.name] smashes against \the [src.name] with its paws.</span>", null, null, COMBAT_MESSAGE_RANGE)
-		take_damage(4, BRUTE, "melee", 1)
```
Comment (yoyobatty) [inline code/game/machinery/_machinery.dm:318]: you change 1 which is true to FALSE, why?

**Comment by MrEpicSax** [comment]
As a science main and a retard I **strongly** approve of this.

**Pull request merged**